### PR TITLE
tools: add log4j example code

### PR DESCRIPTION
**Pull request by PettitWesley**
Signed-off-by: Wesley Pettit <wppttt@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: troubleshooting/tools/log4j-tcp-app/src/main/java/com/mycompany/app/run.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+docker run -it  -e "LOGGER_THREADS=1" -e "LOGGER_ITERATIONS=1" -e "LOG_SIZE_BYTES=1" -e "LOGGER_SLEEP_MS=1000" -e "LOGGER_PORT=5171" -e "INFINITE=true"  log4j:latest
+
+
```
Comment (zhonghui12) [inline troubleshooting/tools/log4j-tcp-app/src/main/java/com/mycompany/app/run.txt:3]: nit: extra line

File: troubleshooting/tools/log4j-tcp-app/src/main/java/com/mycompany/app/App.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
+package com.mycompany.app;
+
+//import org.apache.log4j.Logger;
```
Comment (zhonghui12) [inline troubleshooting/tools/log4j-tcp-app/src/main/java/com/mycompany/app/App.java:3]: There are some codes commented in this file. What are they for? Should we delete them?

File: troubleshooting/tools/log4j-tcp-app/src/main/java/com/mycompany/app/App.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+{
+    private static final Logger logger = LogManager.getLogger(App.class);
+    private static int LOOP_ITERATIONS = 40;
+    private static int INNER_LOOP_ITERATIONS = 12500;
+    private static int LOG_PADDING = 20000;
+
+//	static org.apache.log4j.Logger log = org.apache.log4j.Logger.getLogger(App.class.getName());
+    public static void main( String[] args ) throws InterruptedException
+    {
+    	String tmp;
```
Comment (zhonghui12) [inline troubleshooting/tools/log4j-tcp-app/src/main/java/com/mycompany/app/App.java:26]: nit: extra space before `String`

File: troubleshooting/tools/log4j-tcp-app/src/main/java/com/mycompany/app/App.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        
+        double[] metrics = new double[LOOP_ITERATIONS];
+        double total = 0;
+        long total_ms = 0;
+        
+        
+        for (int i=0; i < LOOP_ITERATIONS; i++) {
+        
+	        long startTime = System.currentTimeMillis();
+
```
Comment (zhonghui12) [inline troubleshooting/tools/log4j-tcp-app/src/main/java/com/mycompany/app/App.java:65]: nit: extra line

File: troubleshooting/tools/log4j-tcp-app/src/main/java/com/mycompany/app/App.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+	        System.out.println("Total execution time: " + (endTime - startTime)  + "ms");
+	        System.out.println("_____________");
+	        java.util.concurrent.TimeUnit.SECONDS.sleep(2);
+        }
+        System.out.println("AVERAGE RATE: " + (total / LOOP_ITERATIONS) + " logs/ms");
+        System.out.println("AVERAGE RATE good math: " + ((LOOP_ITERATIONS * INNER_LOOP_ITERATIONS * 2)/ total_ms) + " logs/ms");
+       
+        double stdev = calculateStandardDeviation(metrics);
+        System.out.println("STDEV: " + stdev + " logs/ms");
+
```
Comment (zhonghui12) [inline troubleshooting/tools/log4j-tcp-app/src/main/java/com/mycompany/app/App.java:97]: extra line

**Review by zhonghui12 — APPROVED** [review]
Thanks Wesley. Added some comments for style issues.

**Review by DrewZhang13 — COMMENTED** [review]
I don't have much functional comments on this PR, mainly about the format and coding style. 

1. Indent need to align inside one class. 
2. unused comments need to be removed.
3. Have a doc comment` /**   */` on above of each class to tell what this class is for.
4. try not to put two class into one file.
5. have some util class to collect the functions like `createPadding` that can be reused everywhere.

**Review by DrewZhang13 — APPROVED** [review]

**Comment by matthewfala** [comment]
Could you add a sample task definition as well?

**Comment by PettitWesley** [comment]
So I'm just gonna merge now this since the point here was just to store some example log4j code for future reference, so we remember how to create an app like this, not to build something perfect and clean that we should actually use in prod or anything like that.

**Pull request merged**